Fix some imports and added some changes to K8S deployment

ojhughes opened the pull request.

Commit:
Fix some imports and added some changes to K8S deployment

scottfrederick commented on deploy/kubernetes/deployment.yml:
```diff
@@ -16,7 +16,7 @@ spec:
     spec:
       containers:
-      - image: sample/bookstore-service-broker:0.0.1.BUILD-SNAPSHOT
+      - image: eu.gcr.io/cf-sandbox-ohughes/bookstore-service-broker:0.0.1.BUILD-SNAPSHOT
```
This change should not be committed

scottfrederick commented on deploy/kubernetes/README.adoc:
```diff
@@ -481,4 +467,19 @@ uri:       67 bytes
+To test the Service Instance API, the URL can be constructed from the binding credentials;
+
+---
+export SI_URI=$(kubectl get secret bookstore-binding -n test -o json | jq -r '.data.uri' | base64 --decode)
```
If we're going to suggest this command, we'll need to add `jq` as an additional pre-requisite at the top of the document. We shouldn't assume everyone will have that installed.

scottfrederick commented on deploy/kubernetes/deployment.yml:
```diff
@@ -2,7 +2,7 @@ apiVersion: extensions/v1beta1
 metadata:
   name: bookstore-broker
-  namespace: bookstore-broker
+  namespace: service-catalog
```
Why the namespace change? That ripples thru a lot of files and doc references. The namespace `service-catalog` might be confused with the namespace that the actual Service Catalog extension is deployed to, or potentially clash with other service broker sample deployments. It would be better if the name were specific to `bookstore`.

scottfrederick commented on deploy/kubernetes/service-broker.yml:
```diff
@@ -5,10 +5,10 @@ metadata:
     run: bookstore-broker
 spec:
-  url: http://bookstore-broker.bookstore-broker.svc.cluster.local
+  url: http://35.197.249.208
```
Will the DNS name not work with `LoadBalanced`? This IP address won't be valid for all deployments, will it?

ojhughes replied:
Yes using the internal DNS works for me

scottfrederick commented on deploy/kubernetes/README.adoc:
```diff
@@ -227,7 +213,7 @@ Now that the application has been deployed and verified, it can be registered to
 
 ----
-$ kubectl create -f deploy/kubernetes/service-broker-secret.yml
+$ kubectl create secret generic -n bookstore-broker bookstore-broker-secret --from-literal=username='admin' --from-literal=password='supersecret'
```
I like the consistency of creating all artifacts from files, but I'm OK with creating with CLI options instead. This commit should remove `deploy/kubernetes/service-broker-secret.yml` if it's not going to be used though.

scottfrederick commented on deploy/kubernetes/README.adoc:
```diff
@@ -171,50 +178,29 @@ Show the details of the service broker application service again:
 
 ----
-$ curl http://192.168.1.235:32248/v2/catalog -u admin:supersecret
+$ curl http://35.197.249.208/v2/catalog -u admin:supersecret
```
Will this IP address be valid for all deployments?

ojhughes replied:
No but I can't see a way of making an IP deterministic without creating and external DNS record

scottfrederick commented on deploy/kubernetes/README.adoc:
```diff
@@ -171,50 +178,29 @@ Show the details of the service broker application service again:
-----
-
-Note the value in the `EXTERNAL-IP` column of the node whose `NAME` matches the node for the `bookstore-broker` pod.
+Note the value in the `LoadBalancer Ingress` column and the `Port` column below.
```
There's no output below this with the specified columns.

scottfrederick commented on deploy/kubernetes/README.adoc:
```diff
@@ -128,18 +135,18 @@ service "bookstore-broker" created
 ----
-$ kubectl describe service bookstore-broker -n bookstore-broker
-Name:                     bookstore-broker
+$ kubectl describe service bookstore-broker -n bookstore-brokerName:                     bookstore-broker
```
Need a line break before `Name: bookstore-broker`

scottfrederick commented:
@ojhughes I had to fix the imports to get another change in. You should be able to rebase and remove those changes.

Commit:
Changes from review -
Ensure internal DNS and correct NS used
Use substituted external IP
Advise on JQ prerequisite
